src/python/WMComponent/PhEDExInjector/PhEDExInjectorPoller.py

```diff
@@ -373,50 +424,224 @@ def recoverInjectedFiles(self):
+                numberOfBlocks += len(blocksToDelete[dataset])
+
+                if numberOfBlocks > chunkSize:
+
+                    self.deleteBlocksPhEDExCalls(location, blocksToDelete)
+                    numberOfBlocks = 0
+                    blocksToDelete = {}
+
+            self.deleteBlocksPhEDExCalls(location, blocksToDelete)
+
+        return
+
+    def deleteBlocksPhEDExCalls(self, location, blocksToDelete):
         """
-        myThread = threading.currentThread()
+        _deleteBlocksPhEDExCalls_
+        actual PhEDEx calls for block deletion
+        """
+        deletion = PhEDExDeletion(blocksToDelete.keys(), location,
+                                  level = 'block',
+                                  comments = "WMAgent blocks auto-delete from %s" % location,
+                                  blocks = blocksToDelete)
+
+        xmlData = XMLDrop.makePhEDExXMLForBlocks(self.dbsUrl,
+                                                 deletion.getDatasetsAndBlocks())
+        logging.debug("deleteBlocks XMLData: %s", xmlData)
+
         try:
-            logging.info("Running PhEDEx injector poller algorithm...")
-            if len(self.blocksToRecover) > 0:
-                logging.info(""" Running PhEDExInjector Recovery: 
-                                 previous injection call failed, 
-                                 check if files were injected to PhEDEx anyway""")
-                self.recoverInjectedFiles()
-
-            self.injectFiles()
-            self.closeBlocks()
-        except PhEDExInjectorPassableError:
-            logging.error("Encountered PassableError in PhEDExInjector")
-            logging.error("Rolling back current transaction and terminating current loop, but not killing component.")
-            if getattr(myThread, 'transaction', None):
-                myThread.transaction.rollbackForError()
-        except Exception:
-            # Guess we should roll back if we actually have an exception
-            if getattr(myThread, 'transaction', None):
-                myThread.transaction.rollbackForError()
-            raise
+            response = self.phedex.delete(deletion, xmlData)
+            requestId = response['phedex']['request_created'][0]['id']
+            # auto-approve deletion request
+            self.phedex.updateRequest(requestId, 'approve', location)
+        except HTTPException as ex:
+            logging.error("PhEDEx block delete/approval failed with HTTPException: %s %s", ex.status, ex.result)
+        except Exception as ex:
+            logging.error("PhEDEx block delete/approval failed with Exception: %s", str(ex))
+            logging.debug("Traceback: %s", str(traceback.format_exc()))
+        else:
+            binds = []
+            for dataset in blocksToDelete:
+                for blockName in blocksToDelete[dataset]:
+                    binds.append( { 'DELETED' : 1,
+                                    'BLOCKNAME' : blockName } )
```

Comment:
These spaces, inside the dict and inside the parenthesis. Also in other places like this one